### DISCUSSION TITLE
Çokdilli Deprem, Afet Yönetimi ve Arama-Kurtarma Terimcesi - Yazılım desteğine ihtiyaç var!

### DISCUSSION THREAD

**Discussion by abdullahsadikkuralay** [acikkaynak/deprem-yardim-projesi#61] [Show and tell]
Merhaba, 

Boğaziçi Üniversitesi Çeviribilim öğrencileri adına yazıyorum. Sahada görevli gönüllü çevirmenler ve deprem bölgesinde farklı dilleri konuşan insanların iletişimini kolaylaştırma amacıyla Google Sheets üzerinde 18 dilde bir arama-kurtarma ve deprem terimcesi oluşturduk. Projemizi yeni terimler ve diller ekleyerek sürekli olarak güncellemekteyiz.

Çevirmenlerin çevrimdışı kullanabileceği, az şarj harcayan, yeni terimler eklendikçe bilgileri Google Sheets'ten çekecek ve kullanıcı dostu arama özellikleri olan bir iOS ve Android uygulaması ihtiyacımız var. Bu konuda nasıl destek alabiliriz?

**Proje linki:** https://docs.google.com/spreadsheets/d/1LfFJZxpbCbz2lRpKQ0_kTMKNJVDJ9FyrlAoP15TU8wU/edit?usp=sharing

**Comment by ramazansancar** [comment]
Merhaba arkadaşlar,
Öncelikle fikriniz için teşekkürler. Bu talebiniz PM ekibine iletildi.

**Reply by hakandilek** [reply]
ben boyle bir ucundan basladim: https://github.com/hakandilek/afet-tercume
su anda angular+pwa olarak hizlica deploy etmeyi dusunuyorum.

gerekirse repoyu tasiyabilir veya paylasabilirim. maksat bir an once bir seyler ciksin.

**Reply by fikret-buyukisik** [reply]
yardımcı olabilirim. Angular, typescript tecrübem var

**Reply by hakandilek** [reply]
@fikret-buyukisik-bam projeye ekledim. bir zoom yapip konusalim musait oldugunuzda

**Reply by hakandilek** [reply]
Gelistirici kaynaklarini wikiye ekledim https://github.com/hakandilek/afet-tercume/wiki

**Comment by TolgaTaner** [comment]
Teknik olarak native planlanıyorsa mobil tarafta iOS kısmına destek olurum - Tolga Taner - iOS#3606

**Comment by hasbisevinc** [comment]
android native düşünülüyorsa yardımcı olabilirim - Discord id:hasbi#1870

**Reply by hasbisevinc** [reply]
Bulgularımı buradan paylaşıyorum:

Auth gerektirmeyen fakat api key ile çalışan googlesheet apisi var
kotaları yeterli görünüyor:
https://developers.google.com/sheets/api/limits

örnek sorgu:
https://sheets.googleapis.com/v4/spreadsheets/1LfFJZxpbCbz2lRpKQ0_kTMKNJVDJ9FyrlAoP15TU8wU/values/A1:Z?key=KEY_BURAYA_GELECEK

daha sonra json parse ederek dillere ulaşılabilir.
Nr ile başlayan satırda diller oluyor. bu kısma yeni diller eklenebiliyor
altındaki satırlar dillerle aynı sırada devam ediyor

**Reply by hakandilek** [reply]
dakikada 300, saniyede 5 request siniri var. sinirlayici olabilir diye dusunuyorum.

**Reply by hasbisevinc** [reply]
@hakandilek , araya belki cache yapacak bir server konabilir bu durumda veya koda birkaç tane api key koyup random olarak birisini kullanması sağlanabilir. datalar çekildikten sonra offline çalışacağı için limit yeterli de olabilir

sizin çalışmanız tamamlandığında offline çalışabilen bir mobil uygulama mı meydana gelecek? Eğer öyleyse native app yapılmasına gerek kalmayacak diye anlıyorum doğru mu?

**Reply by hakandilek** [reply]
mobilde bir PWA yapiyoruz su anda. web sayfasi olarak yuklenecek. data localde saklanacak.

zaten data pek guncellenmedigi icin her seferinde cekmek gerekmiyor.

**Reply by hakandilek** [reply]
@hasbisevinc eger angular uygulamasini ionic gibi bir framework ile Androide cevirmek isterseniz onu da yapabiliriz. playstore/appstore isleri guncellemede zahmetli olur, bizi yavaslatir dusuncesiyle hic girismedim ona.

**Reply by hasbisevinc** [reply]
data localde saklanacaksa mobil uygulamaya gerek kalmıyor diye anlıyorum. fakat excel sürekli güncelleniyor yeni dil ve çeviriler ekleniyor, data refresh butonu koymak faydalı olabilir. 

iyi çalışmalar kolay gelsin

**Reply by TolgaTaner** [reply]
@hasbisevinc  https://github.com/acikkaynak/deprem-yardim-projesi/discussions/35#discussioncomment-4950320 ilgilenirsen burası var
